tests/e2e/image_hotplug_test.go

```diff
@@ -0,0 +1,423 @@
+			})
+		})
+
+		It("compares the disk count before and after attachment", func() {
+			diskCountBefore := len(disksBefore.BlockDevices)
+			Eventually(func() (int, error) {
+				err := GetDisksMetadata(vmObj.Name, &disksAfter)
+				if err != nil {
+					return unacceptableCount, err
+				}
+				diskCountAfter := len(disksAfter.BlockDevices)
+				return diskCountAfter, nil
+			}).WithTimeout(Timeout).WithPolling(Interval).Should(Equal(diskCountBefore+imgCount), "comparing error: 'after' must be equal 'before + %d'", imgCount)
+		})
+
+		It("checks that the `ISO` image is attached as `CD-ROM`", func() {
+			var (
+				isoBlockDeviceName string
+				isolockDeviceCount int
+			)
+			intVirtVmi := &virtv1.VirtualMachineInstance{}
+			err := GetObject(kc.ResourceKubevirtVMI, vmObj.Name, intVirtVmi, kc.GetOptions{
+				Namespace: conf.Namespace,
+			})
+			Expect(err).NotTo(HaveOccurred(), "failed to get `InternalVirtualMachineInstance`: %s", err)
+			for _, disk := range intVirtVmi.Spec.Domain.Devices.Disks {
+				if disk.CDRom != nil {
+					isoBlockDeviceName = disk.Name
+					isolockDeviceCount += 1
+				}
+			}
+			Expect(isolockDeviceCount).To(Equal(1), "there is only one `ISO` block device in this case")
+			isCdRom, err := IsBlockDeviceCdRom(vmObj.Name, isoBlockDeviceName)
+			Expect(err).NotTo(HaveOccurred(), "failed to get `BlockDeviceType` of %q: %s", isoBlockDeviceName, err)
+			Expect(isCdRom).Should(BeTrue(), "wrong type of the block device: %s", isoBlockDeviceName)
+		})
+
+		It("check that the images are attached as the `ReadOnly` devices", func() {
+			imgs := make(map[string]string, imgCount)
+
+			cviImgs := &virtv2.ClusterVirtualImageList{}
+			err := GetObjects(virtv2.ClusterVirtualImageResource, cviImgs, kc.GetOptions{
+				Labels: testCaseLabel,
+			})
+			Expect(err).NotTo(HaveOccurred(), "failed to get `ClusterVirtualImages`: %s", err)
+			for _, cvi := range cviImgs.Items {
+				diskName := fmt.Sprintf("cvi-%s", cvi.Name)
+				imgs[diskName] = ""
+			}
+
+			viImgs := &virtv2.VirtualImageList{}
+			err = GetObjects(virtv2.VirtualImageResource, viImgs, kc.GetOptions{
+				Labels:    testCaseLabel,
+				Namespace: conf.Namespace,
+			})
+			Expect(err).NotTo(HaveOccurred(), "failed to get `VirtualImages`: %s", err)
+			for _, vi := range viImgs.Items {
+				diskName := fmt.Sprintf("vi-%s", vi.Name)
+				imgs[diskName] = ""
+			}
+
+			intVirtVmi := &virtv1.VirtualMachineInstance{}
+			err = GetObject(kc.ResourceKubevirtVMI, vmObj.Name, intVirtVmi, kc.GetOptions{
+				Namespace: conf.Namespace,
+			})
+			Expect(err).NotTo(HaveOccurred(), "failed to get `InternalVirtulMachineInstance`: %s", err)
+			for _, disk := range intVirtVmi.Spec.Domain.Devices.Disks {
+				if _, ok := imgs[disk.Name]; ok {
+					if strings.HasSuffix(disk.Name, "iso") {
+						imgs[disk.Name] = fmt.Sprintf("%s-%s", CdRomIdPrefix, disk.Name)
+					} else {
+						imgs[disk.Name] = fmt.Sprintf("%s_%s", DiskIdPrefix, disk.Serial)
+					}
+				}
+			}
```

Comment:
Listing and populating the imgs map is still unnecessary. This results in extra API calls. Instead, you can directly check if disk.Name has the cvi- or vi- prefix while iterating over intVirtVmi.Spec.Domain.Devices.Disks and construct diskId on the fly.
```suggestion
			imgs := make(map[string]string, imgCount)
			intVirtVmi := &virtv1.VirtualMachineInstance{}
			err = GetObject(kc.ResourceKubevirtVMI, vmObj.Name, intVirtVmi, kc.GetOptions{
				Namespace: conf.Namespace,
			})
			Expect(err).NotTo(HaveOccurred(), "failed to get `InternalVirtulMachineInstance`: %s", err)
			for _, disk := range intVirtVmi.Spec.Domain.Devices.Disks {
				if strings.HasPrefix(disk.Name, "cvi-") || strings.HasPrefix(disk.Name, "vi-") {
					if strings.HasSuffix(disk.Name, "iso") {
						imgs[disk.Name] = fmt.Sprintf("%s-%s", CdRomIdPrefix, disk.Name)
					} else {
						imgs[disk.Name] = fmt.Sprintf("%s_%s", DiskIdPrefix, disk.Serial)
					}
				}
			}
```

Reply:
done